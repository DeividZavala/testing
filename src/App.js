import React, { useState } from "react";
import "./App.css";

function App() {
  const [count, setCount] = useState(0);
  const decrease = () => setCount((prev) => prev - 1);
  const increase = () => setCount((prev) => prev + 1);
  return (
    <div className="App">
      <h1>{count}</h1>
      <button onClick={increase}>+</button>
      <button onClick={decrease}>-</button>
    </div>
  );
}

export default App;
