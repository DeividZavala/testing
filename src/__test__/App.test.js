import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { create } from "react-test-renderer";
import App from "../App";

describe("App test", () => {
  it("should render", () => {
    const { getByRole, getAllByRole } = render(<App />);
    const count = getByRole("heading", { name: "0" });
    const buttons = getAllByRole("button");
    expect(count).toBeInTheDocument();
    expect(buttons.length).toBe(2);
  });
  it("should increment", () => {
    render(<App />);
    const count = screen.getByRole("heading");
    let result = Number(count.textContent) + 1;
    const button = screen.getByRole("button", { name: "+" });
    fireEvent.click(button);
    expect(count.textContent).toBe(`${result}`);
  });
  it("should decrement prro", () => {
    render(<App />);
    const count = screen.getByRole("heading");
    let result = Number(count.textContent) - 1;
    const button = screen.getByRole("button", { name: "-" });
    fireEvent.click(button);
    expect(count.textContent).toBe(`${result}`);
  });
  it("should match snapshot", () => {
    const result = create(<App />).toJSON();
    expect(result).toMatchSnapshot();
  });
  it("should match snapshot on increment", () => {
    const { container } = render(<App />);
    const button = screen.getByRole("button", { name: "+" });
    fireEvent.click(button);
    const result = create(container.innerHTML).toJSON();
    expect(result).toMatchSnapshot();
  });
});
