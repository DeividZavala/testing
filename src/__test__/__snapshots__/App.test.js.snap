// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test should match snapshot 1`] = `
<div
  className="App"
>
  <h1>
    0
  </h1>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;

exports[`App test should match snapshot on increment 1`] = `"<div class=\\"App\\"><h1>1</h1><button>+</button><button>-</button></div>"`;
